File: go/meetup/meetup.go
package meetup

import "time"

// Define the WeekSchedule type here.
type WeekSchedule string

const (
	First WeekSchedule 	= "first"
	Second WeekSchedule	= "second"
	Third WeekSchedule = "third"
	Fourth WeekSchedule = "fourth"
	Fifth WeekSchedule = "fifth"
	Last WeekSchedule = "last"
	Teenth WeekSchedule = "teenth"
)

func Day(wSched WeekSchedule, wDay time.Weekday, month time.Month, year int) int {
	dayCount := 0
	lastDay := -1

	for i := 1; i <= 32; i++ {
		date := time.Date(year, month, i, 0, 0, 0, 0, time.UTC)
		if date.Weekday() != wDay || date.Month() != month {
			continue
		}
		dayCount++
		lastDay = i

		switch {
			case wSched == "first" && dayCount == 1: 
				return i
			case wSched == "second" && dayCount == 2: 
				return i
			case wSched == "third" && dayCount == 3: 
				return i
			case wSched == "fourth" && dayCount == 4: 
				return i
			case wSched == "fifth" && dayCount == 5: 
				return i
			case wSched == "teenth" && i >= 13 && i <= 19:
				return i
		}
	}
	return lastDay
}

File: javascript/meetup/meetup.js
//
// This is only a SKELETON file for the 'Meetup' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

const DAYS_OF_WEEK = [
	"sunday",
	"monday",
	"tuesday",
	"wednesday",
	"thursday",
	"friday",
	"saturday",
];

const toDayOfWeek = (day) => DAYS_OF_WEEK.indexOf(day.toLowerCase());

const isTeenthDay = (day) => day >= 13 && day <= 19;

export const meetup = (year, month, week, day) => {
	const dayOfWeek = toDayOfWeek(day);

	let occurence = 0;
	for (let i = 1; i <= 31; i++) {
		const date = new Date(year, month - 1, i);

		if (date.getDay() === dayOfWeek) {
			occurence++;

			if (occurence === 1 && week === "first") return date;
			if (occurence === 2 && week === "second") return date;
			if (occurence === 3 && week === "third") return date;
			if (occurence === 4 && week === "fourth") return date;
			if (
				new Date(year, month - 1, i + 7).getMonth() !== month - 1 &&
				week === "last"
			)
				return date;
			if (isTeenthDay(date.getDate()) && week === "teenth") return date;
		}
	}
};
